libs/minizip/Makefile

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (C) 2020 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=minizip
+PKG_VERSION:=2.9.3
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/nmoinvaz/minizip/archive/
+PKG_HASH:=f64b2d228a03673a8ec36a53402e2108437226fd9170d49b7f0c0bca94708b93
+
+PKG_MAINTAINER:=David Woodhouse <dwmw2@infradead.org>
+PKG_LICENSE:=Zlib
+PKG_LICENSE_FILES:=LICENSE
+
+CMAKE_INSTALL:=1
+PKG_BUILD_PARALLEL:=1
+
+CMAKE_OPTIONS:=-DINSTALL_INC_DIR=/usr/include/minizip -DMZ_BZIP2=OFF -DBUILD_SHARED_LIBS=yes
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/minizip
+  TITLE:=Fork of the popular zip manipulation library found in the zlib distribution
+  SECTION:=libs
+  CATEGORY:=Libraries
+  DEPENDS:=+zlib
+  URL:=https://github.com/nmoinvaz/minizip
+endef
+
+define Package/minizip-dev
+  SECTION:=devel
+  CATEGORY:=Development
+  SUBMENU:=Libraries
+  DEPENDS:=minizip
+  TITLE:=Development files for the minizip library
+endef
+
+
+define Package/minizip/description
+  minizip is a zip manipulation library written in C that is supported on Windows, macOS, and Linux
+endef
+
+define Package/minizip-dev/description
+  minizip is a zip manipulation library written in C that is supported on Windows, macOS, and Linux
+  This package includes the development support files.
+endef
+
+define Build/InstallDev
+	mkdir -p $(1)/usr/include/
+	$(CP)   $(PKG_INSTALL_DIR)/usr/include/minizip \
+		$(1)/usr/include/
+	mkdir -p $(1)/usr/lib
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/libminizip.so* \
+		$(1)/usr/lib/
+	mkdir -p $(1)/usr/lib/pkgconfig
+	$(CP)   $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/minizip.pc \
```

Comment:
One problem I've seen with CMake packages is faulty pkgconfig file handling. See https://github.com/openwrt/packages/blob/master/libs/pugixml/Makefile#L47 for what I mean. Make sure the pkgconfig files are set up correct. That is, no libdir=/usr/lib or such.

Basically, I have to add that InstallDev section instead of just using CMAKE_INSTALL.

Reply:
So... it's OK for `prefix` and `exec_prefix` to be set explicitly to `/usr` (because they're overridden?) but other things like `includedir` have to be specified in terms of `${prefix}` instead of absolute paths?

Reply:
Absolute paths can be used, but they cannot point to the host. pkgconf is more strict about this than pkg-config.